pkg/rancher-desktop/main/deploymentProfiles.ts

```diff
@@ -126,31 +141,63 @@ function readProfileFiles(rootPath: string, defaultsPath: string, lockedPath: st
  * Windows only. Read settings values from registry using schemaObj as a template.
  * @param schemaObj the object used as a template for navigating registry.
  * @param regKey the registry key obtained from nativeReg.openKey().
- * @param regPath the path to the object relative to regKey.
- * @returns undefined, or the registry data as an object.
+ * @returns null, or the registry data as an object.
  */
-function readRegistryUsingSchema(schemaObj: any, regKey: nativeReg.HKEY, regPath: string[]): any {
-  let regValue;
-  let newObject: any;
+function readRegistryUsingSchema(schemaObj: any, regKey: nativeReg.HKEY): RecursivePartial<settings.Settings>|null {
+  let regValue: any = null;
+  let newObject: RecursivePartial<settings.Settings>|null = null;
 
-  for (const [schemaKey, schemaVal] of Object.entries(schemaObj)) {
-    if (typeof schemaVal === 'object' && !Array.isArray(schemaVal)) {
-      regValue = readRegistryUsingSchema(schemaVal, regKey, regPath.concat(schemaKey));
-    } else {
-      regValue = nativeReg.getValue(regKey, regPath.join('\\'), schemaKey);
-    }
+  const schemaKeys = Object.keys(schemaObj);
+  // ignore case
+  const registryKeys = nativeReg.enumKeyNames(regKey).concat(nativeReg.enumValueNames(regKey)).map(k => k.toLowerCase());
+  const commonKeys = schemaKeys.filter(k => registryKeys.includes(k.toLowerCase()));
```

Comment:
I guess that got lost in order to actually return the values that were in the registry

Reply:
We talked about this, and I said that having unrecognized keys is not an error, just like having extra fields in the JSON file is not treated as an error.

What should be treated as an error is when the types don't match. E.g. when we expect a number, and the value is of type REG_SZ.

One reason not to break on unrecognized keys is that we will add additional settings in the future, that the admin wants to include in their profile. Rolling this out would break RD for user still on an older version because it doesn't know about the new settings.